src/libs/SidebarUtils.js

```diff
@@ -0,0 +1,323 @@
+    callback: val => betas = val,
+});
+
+const lastReportActions = {};
+const reportActions = {};
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+    callback: (actions, key) => {
+        if (!key || !actions) {
+            return;
+        }
+        const reportID = CollectionUtils.extractCollectionItemID(key);
+        lastReportActions[reportID] = _.last(_.toArray(actions));
+        reportActions[key] = actions;
+    },
+});
+
+let policies;
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.POLICY,
+    waitForCollectionCallback: true,
+    callback: val => policies = val,
+});
+
+let iouReports;
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.REPORT_IOUS,
+    waitForCollectionCallback: true,
+    callback: val => iouReports = val,
+});
+
+let currentUserLogin;
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: val => currentUserLogin = val,
+});
+
+let preferredLocale;
+Onyx.connect({
+    key: ONYXKEYS.NVP_PREFERRED_LOCALE,
+    callback: val => preferredLocale = val || CONST.DEFAULT_LOCALE,
+});
+
+/**
+ * @returns {String[]} An array of reportIDs sorted in the proper order
+ */
+function getOrderedReportIDs() {
+    const hideReadReports = priorityMode === CONST.PRIORITY_MODE.GSD;
+    const sortByTimestampDescending = priorityMode !== CONST.PRIORITY_MODE.GSD;
+
+    let recentReportOptions = [];
+    const pinnedReportOptions = [];
+    const iouDebtReportOptions = [];
+    const draftReportOptions = [];
+
+    const filteredReports = _.filter(reports, (report) => {
+        if (!report || !report.reportID) {
+            return false;
+        }
+
+        const isChatRoom = ReportUtils.isChatRoom(report);
+        const isDefaultRoom = ReportUtils.isDefaultRoom(report);
+        const isPolicyExpenseChat = ReportUtils.isPolicyExpenseChat(report);
+        const participants = report.participants || [];
+
+        // Skip this report if it has no participants and if it's not a type of report supported in the LHN
+        if (_.isEmpty(participants) && !isChatRoom && !isDefaultRoom && !isPolicyExpenseChat) {
+            return false;
+        }
+
+        const hasDraftComment = report.hasDraft || false;
+        const iouReport = report.iouReportID && iouReports && iouReports[`${ONYXKEYS.COLLECTION.REPORT_IOUS}${report.iouReportID}`];
+        const iouReportOwner = report.hasOutstandingIOU && iouReport
+            ? iouReport.ownerEmail
+            : '';
+
+        const reportContainsIOUDebt = iouReportOwner && iouReportOwner !== currentUserLogin;
+        const shouldFilterReportIfEmpty = report.lastMessageTimestamp === 0
+
+            // We make exceptions for defaultRooms and policyExpenseChats so we can immediately
+            // highlight them in the LHN when they are created and have no messsages yet. We do
+            // not give archived rooms this exception since they do not need to be higlihted.
+            && !(!ReportUtils.isArchivedRoom(report) && (isDefaultRoom || isPolicyExpenseChat));
```

Comment:
NAB: I added another exception for workspace rooms that failed to add in my refactor PR https://github.com/Expensify/App/pull/10863/. If you merge before I do I can update it, otherwise please update it.
https://github.com/Expensify/App/blob/606677130b13b9d59d6ca9dff81dc9bc9d74607b/src/libs/OptionsListUtils.js#L491-L500

Reply:
OK, I've added this.

I think at some point, I'd really like to simplify all these exceptions. Maybe at least add them to something like `Report.shouldFilterInLHN()`